service/history/ndc/workflow_state_replicator.go

```diff
@@ -226,6 +226,7 @@ func (r *WorkflowStateReplicatorImpl) SyncWorkflowState(
 		namespaceID,
 		wid,
 		rid,
+		branchInfo.GetTreeId(),
```

Comment:
Can you add some comments/TODO here and create an issue to track?

Reply:
Will do